### DISCUSSION TITLE
Add case (capitalize, uppercase, lowercase) option to breadcrumb object

### DISCUSSION THREAD

**Discussion by KBeDevel** [udayvunnam/xng-breadcrumb#32] [Q&A] [answered]
I'm using `xng-breadcrumb@5.0.1` in a project with `angular-cli@9.1.3`. I'd like to be able to set a label with a full uppercase string using a breadcrumb object property

Actually, I get this using CSS in my component:

```css
::ng-deep .xng-breadcrumb-trail {
	text-transform: uppercase;
}
```
But this modify the whole project breacrumb trails to uppercase.

I'm thinking in an optional property named `case` assignable to the breadcrumb object:

```typescript
...,
data: {
  breadcrumb: {
    label: 'foo',
    case: 'capitalize | 'lowercase' | 'uppercase'
  }
}
```

**Comment by KBeDevel** [comment]
**IMPORTANT NOTE**: `breadcrumbService.set()` is not working for this using the versions described above.

**Comment by udayvunnam** [comment]
@KBeDevel you can transform the breadcrumb label as you wish. Please check https://github.com/udayvunnam/xng-breadcrumb#change-text-case.

Let me know if you further need any info

**Comment by udayvunnam** [comment]
Additionally, you will also get the context of first and last items in breadcrumbs for customization. Check - https://github.com/udayvunnam/xng-breadcrumb/blob/730a3091a4f6758d165824e4fb895fdb44dda785/projects/xng-breadcrumb-app/src/app/app.component.html#L33-L41

**Comment by KBeDevel** [comment]
@udayvunnam How can I use string pipes for an specific route param/child?

**Comment by udayvunnam** [comment]
You can pass arbitrary context to a specific route using '**info** param' and that will be passed to **xngBreadcrumbItem**
```javascript
{
	path: 'foo',
	data: {
	  breadcrumb: {
	    label: 'myFoo',
	    info: {case: 'upper'}
	  }
	}
},
{
	path: 'bar',
	data: {
	  breadcrumb: {
	    label: 'myBar',
	    info: {case: 'lower'}
	  }
	}
}
```

```html
<xng-breadcrumb [separator]="separatorTemplate"> 
	<ng-container *xngBreadcrumbItem="let breadcrumb; let info = info> 
       <ng-container *ngIf="info && info.case.upper">{{ breadcrumb | uppercase }}</ng-container> 
       <ng-container *ngIf="info && info.case.lower">{{ breadcrumb | lowercase }}</ng-container> 
       <ng-container *ngIf="!info>{{ breadcrumb }}</ng-container> 
	</ng-container> 
</xng-breadcrumb> 
```

Else, create your own pipe that acts according to the text passed and pass it the info.case text. 
```html
<xng-breadcrumb [separator]="separatorTemplate"> 
	<ng-container *xngBreadcrumbItem="let breadcrumb; let info = info> 
       <ng-container>{{ breadcrumb | myOwnCasePipe: info.case }}</ng-container> 
	</ng-container> 
</xng-breadcrumb> 
```

**Comment by KBeDevel** [comment, marked as answer]
@udayvunnam Nice! It works. But with a little fix. Replace `info.case.upper` with `info.case === 'upper'`

```typescript
...},
{
  path: 'fooPath',
  component: fooComponent
  data: {
    breadcrumb: {
      info: {
        case: 'upper'
      }
    }
  }
},
...
```

```html
<ng-container *xngBreadcrumbItem="let breadcrumb; let info = info">
  <ng-container *ngIf="!info">{{ breadcrumb | titlecase }}</ng-container>
  <ng-container *ngIf="info && info.case === 'upper'">{{ breadcrumb | uppercase }}</ng-container>
  <ng-container *ngIf="info && info.case === 'lower'">{{ breadcrumb | lowercase }}</ng-container>
</ng-container>
```

This worked for me. Thank you.
